### ISSUE TITLE
Duplicate legend entries using plotly and Shiny Server

### ISSUE THREAD

**Issue by nickto** [rstudio/shiny-server#278]
I want to plot data in a Shiny app, using plotly, such that user can select which instances (which factor levels) to plot. The problem is that every legend entry is repeated twice:

![Illustration of duplicate legends](https://cloud.githubusercontent.com/assets/11967792/23921533/44aaeb8e-08ff-11e7-9a05-bc4b28f57276.png)

This issue is only present, when running an app on Shiny Server, i.e. it is **not** present, when running an app locally (or on shinyapps.io):

- [Link](https://nickto.shinyapps.io/reproducible-double-legend-entries/) to this example on shinyapps.io (problem is absent).
- [Link](http://nickto.net/shiny/reproducible-double-legend-entries/) to the same example running on Shiny Server (problem is present).

Reproducible example:

    library(plotly)

    ui <- fluidPage(
      fluidRow(
        column(2, 
               checkboxGroupInput("species", 
                                  label = "Species", 
                                  choices = unique(iris$Species))),
        column(10, 
               plotlyOutput('irisPlotly'))
      )
    )

    server <- function(input, output) {
      # Filter only selected species
      selectedData <- reactive({
        iris[iris$Species %in% input$species, ]
      })

      # Render plotly plot
      output$irisPlotly <- renderPlotly({
        plot_ly(selectedData(), 
                x = ~Sepal.Length, 
                y =~Sepal.Width,
                color = ~Species) %>%
          layout(showlegend = TRUE)
      })
    }

    shinyApp(ui = ui, server = server)

Output of `sessionInfo()` (same behaviour is also present when running Shiny Server under Ubuntu 16.10):

    R version 3.3.2 (2016-10-31) 
    Platform: x86_64-pc-linux-gnu (64-bit) 
    Running under: Arch Linux 

    locale: 
    [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C 
    [3] LC_TIME=en_US.UTF-8 LC_COLLATE=en_US.UTF-8 
    [5] LC_MONETARY=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 
    [7] LC_PAPER=en_US.UTF-8 LC_NAME=C 
    [9] LC_ADDRESS=C LC_TELEPHONE=C 
    [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C 

    attached base packages: 
    [1] stats graphics grDevices utils datasets methods base 

    other attached packages: 
    [1] plotly_4.5.6.9000 ggplot2_2.2.1.9000 shiny_1.0.0 

    loaded via a namespace (and not attached): 
    [1] Rcpp_0.12.9 magrittr_1.5 munsell_0.4.3 viridisLite_0.1.3 
    [5] colorspace_1.3-2 xtable_1.8-2 R6_2.2.0 httr_1.2.1 
    [9] plyr_1.8.4 dplyr_0.5.0 tools_3.3.2 grid_3.3.2 
    [13] gtable_0.2.0 DBI_0.5-1 crosstalk_1.0.0 htmltools_0.3.5 
    [17] yaml_2.1.14 lazyeval_0.2.0 digest_0.6.12 assertthat_0.1 
    [21] tibble_1.2 tidyr_0.6.1 purrr_0.2.2 base64enc_0.1-3 
    [25] htmlwidgets_0.8 mime_0.5 scales_0.4.1 jsonlite_1.3 
    [29] httpuv_1.3.3

**Comment by jcheng5** [comment]
Wow, weird. Is it *precisely* the same build of plotly that's installed on both your Shiny Server server (under the `shiny` account, not whatever account you usually log in as) and locally?

**Comment by nickto** [comment]
You are absolutely right. I had different version of `plotly` for the regular user and for the shiny user. Moreover, the problem is not present in the CRAN version of plotly (`plotly_4.5.6`). Thus, it looks like the issue is present only in the development version of plotly.

Thank you!
